gui/gui/src/TGSpeedo.cxx

```diff
@@ -224,6 +229,16 @@ TGDimension TGSpeedo::GetDefaultSize() const
    return TGDimension(100, 100);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Compute and return the mean of the circular buffer content.
+
+Float_t TGSpeedo::GetMean()
+{
+   if ((fBufferSize == 0) || (fBuffer.size() == 0))
+      return fMeanVal;
+   return (Float_t) std::accumulate(fBuffer.begin(), fBuffer.end(), 0.0) / fBuffer.size();
```

Comment:
Removed a cast, a call to `size()` and  slightly changed the calculation so that it also works in case you call `resize(fBufferSize)` instead of `reserve(fBufferSize)` in `SetBufferSize`.

```suggestion
   return std::accumulate(fBuffer.begin(), fBuffer.begin() + fBufferCount, 0.f) / fBufferCount;
```

Reply:
I don't think this is right. We have to divide by the total buffer size, as we are averaging over N elements. fBufferCount is going cyclically from 0 to BufferSize -1. Thus, one would be dividing by zero very often.
This snippet would be only be correct for the first N times that we fill the buffer. After that, it would yield weird results.
Thus, we need the fBuffer.size() division which correctly tracks how big the 'real data buffer' is.

Reply:
If we would like to optimize, we could define a sticky boolean that is false when filling for the first N times, and is true later on infinitely. One would divide by fBufferCount if bool is false or by fBufferSize if bool is true. But that seems a bit overcomplicating?

Reply:
ah right, nevermind the `+ fBufferCount`